app/assets/javascripts/discourse/app/controllers/tag-show.js

```diff
@@ -1,226 +1,198 @@
-      tracker.resetTracking();
-      return false;
+    @discourseComputed("category", "tag.id", "filterType", "noSubcategories")
+    navItems(category, tagId, filterType, noSubcategories) {
+      return NavItem.buildList(category, {
+        tagId,
+        filterType,
+        noSubcategories,
+        siteSettings: this.siteSettings,
+      });
     },
 
-    changeSort(order) {
-      if (order === this.order) {
-        this.toggleProperty("ascending");
-      } else {
-        this.setProperties({ order, ascending: false });
-      }
-
-      let params = { order, ascending: this.ascending };
-      if (this.period) {
-        params.period = this.period;
-      }
-
-      this.transitionToRoute({
-        queryParams: params,
-      });
+    @discourseComputed("category")
+    showTagFilter() {
+      return true;
     },
 
-    changePeriod(p) {
-      this.set("period", p);
+    @observes("list.canLoadMore")
+    _showFooter() {
+      this.set("application.showFooter", !this.get("list.canLoadMore"));
+    },
 
-      let params = { period: this.period };
+    @discourseComputed("navMode", "list.topics.length", "loading")
+    footerMessage(navMode, listTopicsLength, loading) {
+      if (loading) {
+        return;
+      }
 
-      if (!(this.order === "default" && this.ascending === false)) {
-        params = Object.assign(params, {
-          order: this.order,
-          ascending: this.ascending,
+      if (listTopicsLength === 0) {
+        return I18n.t(`tagging.topics.none.${navMode}`, {
+          tag: this.get("tag.id"),
+        });
+      } else {
+        return I18n.t(`topics.bottom.tag`, {
+          tag: this.get("tag.id"),
         });
       }
-
-      this.transitionToRoute({
-        queryParams: params,
-      });
     },
 
-    toggleInfo() {
-      this.toggleProperty("showInfo");
+    @discourseComputed("list.filter", "list.topics.length")
+    showDismissRead(filter, topicsLength) {
+      return this._isFilterPage(filter, "unread") && topicsLength > 0;
     },
 
-    refresh() {
-      return this.store
-        .findFiltered("topicList", {
-          filter: this.get("list.filter"),
-        })
-        .then((list) => {
-          this.set("list", list);
-          this.resetSelected();
-        });
+    @discourseComputed("list.filter", "list.topics.length")
+    showResetNew(filter, topicsLength) {
+      return this._isFilterPage(filter, "new") && topicsLength > 0;
     },
 
-    deleteTag(tagInfo) {
-      const numTopics =
-        this.get("list.topic_list.tags.firstObject.topic_count") || 0;
-
-      let confirmText =
-        numTopics === 0
-          ? I18n.t("tagging.delete_confirm_no_topics")
-          : I18n.t("tagging.delete_confirm", { count: numTopics });
-
-      if (tagInfo.synonyms.length > 0) {
-        confirmText +=
-          " " +
-          I18n.t("tagging.delete_confirm_synonyms", {
-            count: tagInfo.synonyms.length,
+    actions: {
```

Comment:
can we try to move to @action while we are at it please?

Reply:
Done